spanner.go

```diff
@@ -0,0 +1,179 @@
+package testfixtures
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/googleapis/go-sql-spanner"
+)
+
+type spanner struct {
+	baseHelper
+
+	cleanTableFn func(string) string
+	constraints  []spannerConstraint
+}
+
+type spannerConstraint struct {
+	constraintName   string
+	referencingTable string
+	foreignKeyColumn string
+	referenceTable   string
+	referenceColumn  string
+}
+
+func (h *spanner) init(db *sql.DB) error {
+	var err error
+
+	if h.cleanTableFn == nil {
+		h.cleanTableFn = func(tableName string) string {
+			return fmt.Sprintf("DELETE FROM %s WHERE true;", tableName)
+		}
+	}
+
+	h.constraints, err = h.getConstraints(db)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (*spanner) paramType() int {
+	return paramTypeAtSign
+}
+
+func (*spanner) quoteKeyword(str string) string {
+	return str
+}
+
+func (*spanner) databaseName(q queryable) (string, error) {
+	return "testdb", nil
```

Comment:
there's no easy way to get the database name. How about making the `skipTestDatabaseCheck` mandatory by returning an error here? The message could be "we cannot check whether you're using a test database. Please ensure that you're using a test database and set `skipTestDatabaseCheck` to true" or something similar.

Reply:
could you return an error instead, which indicate that `skipTestDatabaseCheck` is impossible on Spanner, so users need to include `testfixtures.DangerousSkipTestDatabaseCheck()` purposely? 

Also it should be documented in a `README.md`